### DISCUSSION TITLE
Add option to set noindex in mdx file

### DISCUSSION THREAD

**Discussion by spokospace** [withastro/starlight#1916] [Feature Requests]
### What version of `starlight` are you using?

0.22.4

### What is your idea?

It will be good to add additional parameter to add `<meta name="robots" content="noindex" />` to head.

e.g.:

```


---
title: "Title"
description: "Lorem Ipsum"
noindex: true
---


and output example:

``` --
const { title, description, noindex } = Astro.props;
---

<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>{title}</title>
  <meta name="description" content={description} />
  {noindex && <meta name="robots" content="noindex" />}
</head>
<body>
  <slot />
</body>
</html>
```

### Why is this feature necessary?

It is good to have control over the pages, for some pages I need to disable option for indexing by search engine robots.

### Do you have examples of this feature in other projects?

In other projects on Astro I usually use [astro-seo](https://github.com/jonasmerlin/astro-seo) for this and the setting `noindex={true}`.
![image](https://github.com/withastro/starlight/assets/16397823/9af19aec-87af-4360-be60-c448b1d59012)

https://github.com/jonasmerlin/astro-seo/blob/7d3387a2d90aab5fc622def6994520f705c685e3/src/pages/noindex.astro#L9

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
Thank you for the suggestion!

In the meantime you can do this using the [`head` frontmatter field](https://starlight.astro.build/reference/frontmatter/#head):

```md
---
title: Don’t index me!
head:
  - tag: meta
    attrs:
      name: robots
      content: noindex
---

Content...
```
